### DISCUSSION TITLE
How to pass dynamic jwt token in rest client ?

### DISCUSSION THREAD

**Discussion by woodsongem** [quarkusio/quarkus#24119] [Community]
Hello,

we are calling extra customer api using the REST rest client . want to pass dynamic jwt token as part of the request header with some more dynamic token. not sure how to pass. TIA

`
@Path("/api")
@RegisterRestClient
public interface CustomerApiRestClient {
    @GET
    @Path("/loans/{loanNumber}")
    Customer getCustomerDetail(@PathParam Long customerId);
}
`

**Comment by sberyozkin** [comment]
@woodsongem Hi, can you clarify please what you mean by `want to pass dynamic jwt token as part of the request header with some more dynamic token` ?

**Reply by woodsongem** [reply]
we have to generate auth0 token and attached into getCustomerDetail request header bearer token. also to generate Authtoken based on parameter in current request parameters.

**Comment by sberyozkin** [comment]
@woodsongem Perhaps https://quarkus.io/guides/security-openid-connect-client can help ? `OidcClientFilter` can be used to acquire an auth0 token and propagate it. Or the existing token coming in to the endpoint as a bearer token can be propagated directly with the token propagation filter.
And you can generate a new token with https://quarkus.io/guides/security-jwt-build

**Comment by woodsongem** [comment]
thanks @sberyozkin . I am using OidcClientConfig to build new client based on our required . but I didn't find way to set audience in OidcClientConfig

**Reply by woodsongem** [reply]
thanks @sberyozkin . able to pass audience as part of grantOptions and it's working
